airflow/models/connection.py

```diff
@@ -348,6 +348,25 @@ def debug_info(self):
             self.extra_dejson,
         )
 
+    def test_connection(self):
+        """Calls out get_hook method and executes test_connection method on that."""
+        status, message = False, ''
+        try:
+            hook = self.get_hook()
+            if hook and getattr(hook, 'test_connection', False):
+                status, message = hook.test_connection()
+            else:
+                message = (
+                    f"Hook {hook.__class__.__name__} doesn't implement or inherit test_connection method"
```

Comment:
This could also be no hook class found.

Reply:
yeah, in that the message would be `Hook NoneType doesn't...`

Reply:
Which is a bit confusing to a user if it happens -- I think we should have a different message for that case.